tests/API/Management/UsersTest.php

```diff
@@ -4,61 +4,205 @@
+        $this->assertEquals($this->getId($paged_results[0]), $this->getId($many_results[$page_num]));
+
+        return $many_results;
     }
 
-    protected function getCreateBody() {
-        $this->email = 'test-create-user' . rand();
-        echo "\n-- Using user email {$this->email} \n";
+    /**
+     * Check that the User created matches the initial values sent.
+     *
+     * @param array $entity - The created User to check against initial values.
+     */
+    protected function afterCreate($entity)
+    {
+        $expected = $this->getCreateBody();
+        $this->assertNotEmpty($this->getId($entity));
+        $this->assertEquals($expected['email'], $entity['email']);
+        $this->assertFalse($entity['email_verified']);
+        $this->assertEquals($expected['user_metadata']['key1'], $entity['user_metadata']['key1']);
+        $this->assertEquals($expected['user_metadata']['key2'], $entity['user_metadata']['key2']);
+    }
 
+    /**
+     * Get the User update data to send with the test update call.
+     *
+     * @return array
+     */
+    protected function getUpdateBody()
+    {
         return [
-            'connection' => 'Username-Password-Authentication',
-            'email' => $this->email . '@test.com',
-            'password' => '123456',
-            'email_verified' => true
+            'email' => 'test-update-user-' . $this->rand . '@auth0.com',
+            'email_verified' => true,
+            'user_metadata' => [
+                'key1' => 'value4',
+                'key3' => 'value3',
+            ],
         ];
     }
 
-    protected function getUpdateBody() {
-        return ['email' => $this->email . 'changed@test.com'];
-    }
-    protected function afterCreate($entity) {
-        $this->assertEquals($this->email . '@test.com', $entity['email']);
+    /**
+     * Update entity returned values check.
+     *
+     * @param array $entity - User that was updated.
+     */
+    protected function afterUpdate($entity)
+    {
+        $expected = $this->getUpdateBody();
+        $this->assertEquals($expected['email'], $entity['email']);
+        $this->assertTrue($entity['email_verified']);
+        $this->assertEquals($expected['user_metadata']['key1'], $entity['user_metadata']['key1']);
+        $this->assertEquals($expected['user_metadata']['key3'], $entity['user_metadata']['key3']);
     }
-    protected function afterUpdate($entity) {
-        $this->assertEquals($this->email . 'changed@test.com', $entity['email']);
+
+    /**
+     * Test whether the User create function throws errors correctly.
+     */
+    public function testRequiredUserCreateFields()
+    {
+
+        // Try to create a user without a Connection.
+        $caught_connection_error = false;
+        try {
+            $this->api->create([]);
+        } catch (\Exception $e) {
+            $caught_connection_error = $this->errorHasString($e, 'connection');
```

Comment:
a bit vague. I don't remember the full error message but what about searching for something like "missing param 'connection'" at least? Same for the checks below